src/Microsoft.Health.Fhir.Shared.Core/Features/Operations/Everything/PatientEverythingService.cs

```diff
@@ -292,18 +294,9 @@ private async Task<string> CheckForNextSeeAlsoLinkAndSetToken(string parentPatie
             // Retrieve the parent patient so that we can extract its links and process the next "seealso" link.
             using IScoped<ISearchService> search = _searchServiceFactory.Invoke();
 
-            // To do so, first create the search parameter to add to the query
-            var searchParameters = new List<Tuple<string, string>>
-            {
-                Tuple.Create(SearchParameterNames.Id, parentPatientId),
-            };
-
-            // And then execute the search.
-            SearchOptions searchOptions = _searchOptionsFactory.Create(KnownResourceTypes.Patient, searchParameters);
-            SearchResult searchResult = await search.Value.SearchAsync(searchOptions, cancellationToken);
-
-            // The search result entries should contain one patient resource - the parent patient.
-            SearchResultEntry parentPatientResource = searchResult.Results.FirstOrDefault(s => string.Equals(s.Resource.ResourceTypeName, KnownResourceTypes.Patient, StringComparison.Ordinal));
+            // Get the parent patient history and find the version we recorded in the first $everything operation API call.
+            SearchResult searchResult = await search.Value.SearchHistoryAsync(KnownResourceTypes.Patient, parentPatientId, null, null, null, null, null, cancellationToken);
+            SearchResultEntry parentPatientResource = searchResult.Results.FirstOrDefault(s => string.Equals(s.Resource.Version, token.ParentPatientVersionId, StringComparison.Ordinal));
```

Comment:
Is this the most efficient way to retrieve a specific version of a resource?

Reply:
You should be able to fetch the specific version directly by making a ResourceKey like is done here: https://github.com/microsoft/fhir-server/blob/6a4268c9c04f01e3d29ae662782cccd46cc031ec/src/Microsoft.Health.Fhir.Shared.Core/Features/Resources/Get/GetResourceHandler.cs#L44

Reply:
Ah yes perfect, thanks! Implemented in 636fd29.